src/server/api/controllers/messages-controller.js

```diff
@@ -1,10 +1,46 @@
+/* eslint-disable @typescript-eslint/camelcase */
+/* eslint-disable camelcase */
 const knex = require('../../config/db');
 const Error = require('../lib/utils/http-error');
 const moment = require('moment-timezone');
 
-const getMessages = async () => {
+const getMessages = async (query, channel_id, sender, limit, sort, sort_by) => {
+  const getMes = knex('channel_messages')
+    .select(
+      knex.raw(
+        'DISTINCT channel_messages.id, channel_messages.message, channel_messages.fk_channel_id,channel_messages.fk_user_id,channel_messages.created_at, channel_messages.updated_at,channel_messages.deleted_at,users.id,users.profile_image, users.user_name,users.email, users.last_seen, users.created_at, users.updated_at, users.deleted_at',
+      ),
+    )
+    .leftJoin(knex.raw('users ON users.id = channel_messages.fk_user_id'));
   try {
-    return await knex('messages').select('messages.id', 'messages.title');
+    if (query) {
+      return await getMes.where(
+        'channel_messages.message',
+        'like',
+        `%${query}%`,
+      );
+    }
+    if (channel_id) {
+      return await getMes.where('channel_messages.id', channel_id);
+    }
+    if (sender) {
+      return await getMes.where('users.email', 'like', `%${sender}%`);
+    }
+    if (limit) {
+      return await getMes.limit(limit);
+    }
+    if (sort) {
+      return await getMes.orderBy('channel_messages.id', sort);
+    }
+    if (sort_by) {
```

Comment:
What is the difference between sort and sortby?

Reply:
It is response to this task "Sorting: /api/messages?sort=desc&sort-by=date"
"sort" sorting by abc/desc and the sort_by  sorting bay date.
now I changed sort_by to date If anyone want to search for any specific date . What do you think ?